zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -31,7 +31,7 @@
 
     public static final String AUTHPROVIDER_PROPERTY_PREFIX = "zookeeper.authProvider.";
 
-    private static boolean initialized = false;
+    private static volatile boolean initialized = false;
     private static final Map<String, AuthenticationProvider> authenticationProviders = new HashMap<>();
 
```

Comment:
Look at the call/write of `initialized`, `reset()` is only used for testing, and `initialize()` is used at the start-up of server. so we don't have the situation when `initialized` is visited by multi-threads, one thread changes the value and it should be visible at once for other threads?